### ISSUE TITLE
Add Traversable isOrdered() indicator and Ordered interface

### ISSUE THREAD

**Issue by danieldietrich** [javaslang/javaslang#1495]
This comes handy when converting a specific Traversable to an ordered collection like SortedSet or PriorityQueue. We could get rid of using `instanceof`.

``` java
// old code
if (this instanceof SortedSet) {
    Comparator<T> comparator = ((SortedSet<T>) this).comparator();
    ...
} else if (this instanceof PriorityQueue) {
    Comparator<T> comparator = ((PriorityQueue <T>) this).comparator();
    ...
} else if (this instanceof ...) {
    Comparator<T> comparator = ((...<T>) this).comparator();
    ...
} else {
    ...
}

// new code
if (isOrdered()) {
    Comparator<T> comparator = ((Ordered<T>) this).comparator();
    ...
} else {
   ...
}
```

The new `Ordered` interface looks like this:

``` java
interface Ordered<T> {
    Comparator<T> comparator();
}
```

It will be implemented by SortedSet, PriorityQueue, ... The existing `comparator()` methods need an `@Override` annotation then.

**Comment by danieldietrich** [comment]
This issue goes adhere with #1488

**Comment by danieldietrich** [comment]
See also #1635

**Comment by ruslansennov** [comment]
Closed by #1711
